packages/knex/src/AbstractSqlDriver.ts

```diff
@@ -187,18 +249,92 @@ export abstract class AbstractSqlDriver<C extends AbstractSqlConnection = Abstra
   /**
    * 1:1 owner side needs to be marked for population so QB auto-joins the owner id
    */
-  protected autoJoinOneToOneOwner(meta: EntityMetadata, populate: string[]): string[] {
+  protected autoJoinOneToOneOwner(meta: EntityMetadata, populate: PopulateOptions[]): PopulateOptions[] {
     if (!this.config.get('autoJoinOneToOneOwner')) {
       return populate;
     }
 
-    const toPopulate = Object.values(meta.properties)
-      .filter(prop => prop.reference === ReferenceType.ONE_TO_ONE && !prop.owner && !populate.includes(prop.name))
-      .map(prop => prop.name);
+    const relationToPopulate = populate.map(({ field }) => field);
+
+    const toPopulate: PopulateOptions[] = Object.values(meta.properties)
+      .filter(prop => {
+        return prop.reference === ReferenceType.ONE_TO_ONE && !prop.owner && !relationToPopulate.includes(prop.name);
+      })
+      .map((prop) => {
+        return {
+          field: prop.name,
+          strategy: prop.strategy,
+        };
+      });
 
     return [...populate, ...toPopulate];
   }
 
+  protected joinedLoads(meta: EntityMetadata, populate: PopulateOptions[]): string[] {
+    return populate.filter(({ field }) => {
+      const relation = meta.properties[field];
+
+      if (!relation) {
+        return false;
+      }
+
+      return relation.strategy === LoadStrategy.JOINED;
+    }).map(({ field }) => field);
+  }
+
+  protected processJoinedLoads<T extends AnyEntity<T>>(results: object[], joinedLoads: string[]): T {
+    return results.reduce((accumulator, value) => {
+      joinedLoads.forEach((relationName) => {
+        if (relationName in value) {
+          const relation = value[relationName];
+
+          if (Array.isArray(relation)) {
+            const existing = accumulator[relationName] || [];
+            accumulator[relationName] = [...existing, ...relation];
+          }
+        }
+      });
+
+      return {
+        ...value,
+        ...accumulator,
+      };
+    }, {}) as unknown as T;
+  }
+
+  protected getSelectForJoinedLoad(queryBuilder: QueryBuilder, meta: EntityMetadata, joinedLoads: string[]): Field[] {
+    const selects: Field[] = [];
+
+    // alias all fields in the primary table
+    Object.values(meta.properties)
+      .filter(({ reference, persist }) => {
+        return reference === 'scalar' && persist !== false;
+      })
+      .forEach((descriptor) => {
+        const [fieldName] = descriptor.fieldNames;
+
+        selects.push(fieldName);
+        queryBuilder.select(fieldName);
+      });
+
+    joinedLoads.forEach((relationName) => {
+      const prop = meta.properties[relationName];
+      const entity = this.metadata.get(prop.type);
+      const { properties } = entity;
+
+      Object.values(properties)
+        .filter(({ reference, persist }) => reference === 'scalar' && persist !== false)
+        .forEach((descriptor) => {
```

Comment:
we use `prop` everywhere in the code base so please stick to that.
```suggestion
        .forEach(prop => {
```